test/groovy/com/sap/piper/tools/neo/NeoCommandHelperTest.groovy

```diff
@@ -121,9 +122,9 @@ class NeoCommandHelperTest extends BasePiperTest {
 
     @Test
     void deployMta() {
-        String actual = getTestFixture(DeployMode.MTA).deployMta()
+        String actual = getTestFixture(DeployMode.MTA, (Set)['myExtension1.yml', 'myExtension2.yml']).deployMta()
```

Comment:
As far as I understand it we are coercing anyway when calling `def x = (Set)foo` in case `foo` is not a Set. With plain java we would get a `ClassCastException`. But with groovy it gets coerced - like with `foo.asSet()` et al.

We can only _cast_ to another type if `foo` is compatible. Otherwise we can _cast_, but in groovy we can _coerce_.

Correct me if I'm wrong.

Reply:
> As far as I understand it we are coercing anyway
No, we are casting, so when we cast we have to make sure that the runtime type will be correct, e.g using `instanceof`.